src/node.cc

```diff
@@ -4616,6 +4620,29 @@ int Start(int argc, char** argv) {
   return exit_code;
 }
 
+#ifdef __LP64__
+# define Elf_auxv_t Elf64_auxv_t
+#else
+# define Elf_auxv_t Elf32_auxv_t
+#endif
+
+#if defined(__linux__)
+AuxVector::AuxVector() {
+  char** envp = environ;
+  while (*envp++ != NULL) {}
+  Elf_auxv_t* auxv = reinterpret_cast<Elf_auxv_t*>(envp);
+  for (; auxv->a_type != AT_NULL; auxv++) {
+    auxv_map_[auxv->a_type] = auxv->a_un.a_val;
+  }
```

Comment:
You should skip AT_IGNORE entries here.  That said, I'd make this the simplest possible implementation and not bother with a map since we're only interested in AT_SECURE.  E.g. `bool linux_at_secure;` in node.cc in and set it in node_main.cc if AT_SECURE is present (with an `extern bool linux_at_secure;` declaration to make it link.)

I'd be okay with unconditionally defining the bool to save on `#ifdef __linux__` ugliness.